src/main/kotlin/no/nav/familie/ef/sak/service/steg/InngangsvilkårSteg.kt

```diff
@@ -0,0 +1,31 @@
+package no.nav.familie.ef.sak.service.steg
+
+import no.nav.familie.ef.sak.api.Feil
+import no.nav.familie.ef.sak.repository.domain.Behandling
+import no.nav.familie.ef.sak.repository.domain.BehandlingType
+import no.nav.familie.ef.sak.service.VurderingService
+import no.nav.familie.ef.sak.util.RessursUtils.lagFrontendMelding
+import org.springframework.stereotype.Service
+
+@Service
+class InngangsvilkårSteg(private val vurderingService: VurderingService): BehandlingSteg<String> {
+
+    override fun utførStegOgAngiNeste(behandling: Behandling, data: String): StegType {
+        return hentNesteSteg(behandling)
+    }
+
+    override fun stegType(): StegType {
+        return StegType.VILKÅRSVURDERE_INNGANGSVILKÅR
+    }
+
+    override fun postValiderSteg(behandling: Behandling) {
+        if (behandling.type == BehandlingType.TEKNISK_OPPHØR) return
+
+        val vilkårSomManglerVurdering = vurderingService.hentInngangsvilkårSomManglerVurdering(behandling.id)
+
+        if (vilkårSomManglerVurdering.isNotEmpty())
```

Comment:
Tenker dette burde skje i `preValiderSteg`, æven om det ikke skje noe i utførSteg idag?

Reply:
Hva er grunnen til at du mener det? :) Regner med at du mener at hele innmaten her burde flyttes til preValider-funksjonen? Jeg tenker at det ikke gir mening å gjøre denne sjekken før saksbehandler har fått mulighet til å fylle ut vurdering for alle inngangsvilkårene, så dette er det siste som bør gjøres før behandlingen går videre til neste steg. Dermed passer det i en `postValiderSteg`-funksjon. Alternativt kunne vi lagt det inn i selve utførSteg-funksjonen siden det ikke er noe annet der nå, men hvis vi skulle finne på å legge noe annet inn der (som feks automatisk evaluering av noen vilkår), må vi likevel flytte denne sjekken tilbake til postValider-funksjonen.

Reply:
Det er sikkert fordi jeg ikke skjønner interfacet for denne klassen.
3 metoder, som kalles i rekkefølgen 
* preValider
* utfør
* postValider

Hva er forventet å skje i de ulike steget? Jeg renamet de til preUtfør, utfør og postUtfør, og forventer meg att noe skal skje i utfør, men att preUtfør er en form av validering.
Men å legge validering etter att man utført ett steg ser jeg ingen logikk i? Er kanskje enklere hvis noen kan forklare meg hvordan dette flyt skal være, syns det virker veldig ulogisk.

Reply:
Vi tar denne diskusjonen utenfor denne PR